modules/java-modules/kafka/src/main/java/org/syslog_ng/KafkaDestination.java

```diff
@@ -46,8 +46,11 @@
 
   public KafkaDestination(long handle) {
     super(handle);
+
+    Logger.getLogger("org.apache.kafka").setLevel(SyslogNgInternalLogger.getLevel());
```

Comment:
Why do we need to set this logger ? Does this mean that it branches off too early from root logger and not having the proper log level, or that root logger log level is not set according to syslog-ng level ?
Should we set log level for other java based destination's logger?

Reply:
The root log level has been set to the syslog-ng internal's log level in this PR.

> Should we set log level for other java based destination's logger?

I think it's not necessary because we set the root log level and all other child loggers inherit the root (their parent's) level.

Kafka seems to be different, I haven't checked their implementation, but they might set their own log level because ours is not taken into account. I would consider this commit as a workaround.

Reply:
Thanks for the explanation.